pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/strategy/BalancedNumSegmentAssignmentStrategy.java

```diff
@@ -51,6 +51,10 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
     SegmentsValidationAndRetentionConfig validationAndRetentionConfig = tableConfig.getValidationConfig();
     Preconditions.checkState(validationAndRetentionConfig != null, "Validation Config is null");
     _replication = validationAndRetentionConfig.getReplicationNumber();
+    // Number of replicas per partition of low-level consumers check is for the real time tables only
+    if (validationAndRetentionConfig.getReplicasPerPartition() != null) {
```

Comment:
We need to check if the table is offline or real-time and then decide where to pull the replication data.

Reply:
Is it possible to wrap all the logic for whether to use `replication` or `replicasPerPartition` within the TableConfig itself? That way, `tableConfig.getReplicationNumber` should just return the correct value depending on whether it is offline or realtime table.

Reply:
That was the original thought @navina . @snleee I updated the check

Reply:
But we saw real time tables using balanced strategy in Production. @jugomezv can you please confirm

Reply:
@navina We can wire that in table config but it would also mean cleaning up at all the other places we use this. I would so keep that out of scope for this fix

Reply:
Please tag this PR with the (open) issue https://github.com/apache/pinot/issues/8804.

Since we have a problem in production, we should fix in some way (add TODOs as needed), but fix the overall replication/replicasPerPartition as a part of that issue.

Reply:
done